packages/components/src/internal/url/URLResolver.ts

```diff
@@ -513,6 +513,14 @@ export const URL_MAPPERS = {
 
 export class URLResolver {
     private mapURL = (mapper: MapURLOptions): string => {
+
+        // Don't override URLs if the URL has a different container than the current container and is not in the subfolder
```

Comment:
Couple things.

1. We want to make sure this works in both directions. If in the project folder, then this logic should skip any URLs for any folder underneath the project. If in a subfolder, at any depth, then this logic should skip any URLs above/below this subfolder.
2. I see a bunch of test cases have been updated but I didn't notice any that were specifically targeted at these scenarios. We should add test cases to `URLResolver.spec.ts` specifically for this processing.